### PR TITLE
feat: Add Arbitrum Pool

### PR DESCRIPTION
**Pull request by ChefMomota**
### Summary
🌉🍰🚀

This pull request adds support for the Arbitrum One and Arbitrum Goerli networks to the app. It creates and updates the files for the price helper LPs, the pools, the contracts, and the tokens on these networks. It also adds the CAKE and mockA tokens to the default and common token lists. This enables the users to access the features and functionalities of the app on these chains and to interact with the CAKE and mockA tokens.

> _Oh we're the `CAKE` lovers and we sail the Arbitrum sea_
> _We swap and stake our tokens on the network that is free_
> _We've added pools and farms and vaults for every mate to see_
> _So heave away and pull the code, and one, two, three!_

### Walkthrough
*  Add support for the CAKE token, the iCAKE token, the CAKE vault, and the CAKE flexible side vault on the Arbitrum One and Arbitrum Goerli networks ([link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-6ccab19ffbec70686dc2221f7b79432ece4302c36067c91063d258209c09ec07R31-R32), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-6617e1b6b671418fcbaeee30a67fa7198f5b270e377dccf8393cd40db8b92534R14-R15), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-6617e1b6b671418fcbaeee30a67fa7198f5b270e377dccf8393cd40db8b92534R22-R23), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-6617e1b6b671418fcbaeee30a67fa7198f5b270e377dccf8393cd40db8b92534R30-R31))
*  Add the contract addresses for these tokens and contracts to the `contracts.ts` files in the `web` and `pools` packages ([link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-6ccab19ffbec70686dc2221f7b79432ece4302c36067c91063d258209c09ec07R31-R32), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-6617e1b6b671418fcbaeee30a67fa7198f5b270e377dccf8393cd40db8b92534R14-R15), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-6617e1b6b671418fcbaeee30a67fa7198f5b270e377dccf8393cd40db8b92534R22-R23), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-6617e1b6b671418fcbaeee30a67fa7198f5b270e377dccf8393cd40db8b92534R30-R31))
*  Add the CAKE token on the Arbitrum Goerli network to the default token list and the common tokens file ([link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-55ee773d94e57d4fe20f6463bc02307f5d9747933c77b517fa31a90375d2af7eR127-R134), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-8dce334490b3aef54ec2484deec6ef79b4438828dc1ee3324cfa3b2a5244251bR182-R189))
*  Add the mockA token to the list of tokens on the Arbitrum One and Arbitrum Goerli networks ([link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-ab21739cfdac48707d4b9c121d061b2856a86c256319d79879d8a7e5be8c08a8L8-R12), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-37ca7b1aa87e421c6aca20950f5f888ef838e8067a8c098aceab66c1900149f7L1-R8))
*  Add the price helper LP files for the Arbitrum One and Arbitrum Goerli networks to the `priceHelperLps` folder in the `web` package ([link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-d78bf8d1c667faba4e13657e1bbc2abe377978eb57c350b22dfe54e3354c8ab2R7-R8), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-928f23368b88f820da1d7ee2272de90402aa9343ca3cf0134141fdb346d82c9cR1-R5), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-0f4aa3518b3110a7b504142c05ea3af2b775071999a717ea11d08c335159948dR1-R5))
*  Add the cases for these networks to the function that returns the price helper LP files based on the chain ID ([link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-d78bf8d1c667faba4e13657e1bbc2abe377978eb57c350b22dfe54e3354c8ab2R22-R25))
*  Add the pools files for the Arbitrum One and Arbitrum Goerli networks to the `pools` folder in the `pools` package ([link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-6778b4df10ef297ee3b92123573ffd299db118acd388728f8b3d2ef98d9571b1R1-R26), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-6ac78babcc1a76e48c4064cf56e9f95f77af61ea1a431d3f1400295b959437adR1-R69))
*  Add the cases for these networks to the objects that map the chain ID to the pools files and the live pools files ([link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-f5ab55bf4078135a7bd2ebb8bf5c23c7f0de1e8c6d36c27861bc43805419dc25R20-R21), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-f5ab55bf4078135a7bd2ebb8bf5c23c7f0de1e8c6d36c27861bc43805419dc25R28-R29))
*  Import the pools files and the price helper LP files for these networks in the `index.ts` files in the `pools` and `priceHelperLps` folders ([link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-d78bf8d1c667faba4e13657e1bbc2abe377978eb57c350b22dfe54e3354c8ab2R7-R8), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-f5ab55bf4078135a7bd2ebb8bf5c23c7f0de1e8c6d36c27861bc43805419dc25R6-R7))
*  Add the Arbitrum One and Arbitrum Goerli networks to the list of supported chain IDs in the `supportedChains.ts` files in the `web` and `pools` packages ([link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-d5561cd39967a71b1fe70cdb06abed66dd18633d9936a78951c0d6c30dbf905eR14), [link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-9d58fdf9947758c1915460c83ccb4265e39446a1e47804ca27c2500580945d09L3-R9))
*  Add the Arbitrum Goerli network to the list of supported farms in the `supportChains.ts` file in the `web` package ([link](https://github.com/pancakeswap/pancake-frontend/pull/7566/files?diff=unified&w=0#diff-d5561cd39967a71b1fe70cdb06abed66dd18633d9936a78951c0d6c30dbf905eR14))

### CODE REVIEW

File: packages/tokens/src/42161.ts [hunk 1]
```diff
@@ -5,8 +5,9 @@ export const arbitrumTokens = {
   weth: WETH9[ChainId.ARBITRUM_ONE],
   usdt: USDT[ChainId.ARBITRUM_ONE],
   usdc: USDC[ChainId.ARBITRUM_ONE],
+  cake: CAKE[ChainId.ARBITRUM_ONE],
   arb: new ERC20Token(ChainId.ARBITRUM_ONE, '0x912CE59144191C1204E64559FE8253a0e49E6548', 18, 'ARB', 'Arbitrum'),
   gmx: new ERC20Token(ChainId.ARBITRUM_ONE, '0xfc5A1A6EB076a2C7aD06eD22C90d7E710E35ad0a', 18, 'GMX', 'GMX'),
   wbtc: new ERC20Token(ChainId.ARBITRUM_ONE, '0x2f2a2543B76A4166549F7aaB2e75Bef0aefC5B0f', 8, 'WBTC', 'Wrapped BTC'),
-  cake: CAKE[ChainId.ARBITRUM_ONE],
+  mockA: new ERC20Token(ChainId.ARBITRUM_ONE, '0x47B2286A76acF9a99B30c8D8261EF6Eb066dafC7', 18, 'A', 'MOCK Token A'),
```
Comment (chefjackson) [inline packages/tokens/src/42161.ts:12]: let's remove this one on launch

File: apps/web/src/utils/fetchTokenAplPrice.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      },
+    ],
+    allowFailure: false,
+  })
+
+  const alpUsdPrice = getBalanceAmount(new BigNumber(alpPrice.toString()), TOKEN_DECIMALS).toNumber()
+  return alpUsdPrice ?? 0
+}
+
+export default fetchTokenAplPrice
```
Comment (chefjackson) [inline apps/web/src/utils/fetchTokenAplPrice.ts:28]: Let's use named export instead of default export

Comment (ChefMomota) [reply apps/web/src/utils/fetchTokenAplPrice.ts]: done

File: apps/web/src/state/pools/index.ts [hunk 3]
```diff
@@ -210,9 +211,19 @@ export const fetchPoolsPublicDataAsync = (chainId: number) => async (dispatch, g
...
-        const result = await fetchTokenUSDValue(chainId, [stakingTokenAddress])
-        stakingTokenPrice = result.get(stakingTokenAddress) || 0
+        // TODO: Remove this when fetchTokenUSDValue can get APL USD Price
+        if (
+          pool.stakingToken.chainId === ChainId.ARBITRUM_ONE &&
+          pool.stakingToken.address === arbitrumTokens.alp.address
+        ) {
+          // eslint-disable-next-line no-await-in-loop
+          const result = await fetchTokenAplPrice()
+          stakingTokenPrice = result
```
Comment (chefjackson) [inline apps/web/src/state/pools/index.ts:221]: no need for an extra `result` variable?

Comment (chefjackson) [reply apps/web/src/state/pools/index.ts:221]: yeah we can simply write `stakingTokenPrice = await fetchTokenAplPrice()`

Comment (ChefMomota) [reply apps/web/src/state/pools/index.ts]: done

**Review by chefjackson — APPROVED** [review]

**Pull request merged**